### ISSUE TITLE
No longer getting the wineasio to register after upgrading to winehq-stable 9

### ISSUE THREAD

**Issue by SwampGod** [wineasio/wineasio#81]
I can no longer get the Kxstudio wineasio to register after upgrading to winehq-stable 9 or staging 9. I hard coded the dlls in my startup script and still no juice. Just an FYI but a solution would be nice. I will run down the compiling from source road if need be.

**Comment by moonshaadow** [comment]
Hi there, 
Same issue here, `wineasio-register` is not able to register dll.

**Comment by falkTX** [comment]
yeah wine-9 changed some stuff yet again. the DLLs no longer load.

the good news is that I am able to reproduce the issue, which is a good first step.

**Comment by punkochel** [comment]
I was able to register wineasio64.dll under wine 9.0 stable.
We compile from wineasio64 sources. Dependencies: `libwine-development-dev, libjack-jackd2-dev`.
Drop `wineasio64.so` and `wineasio64.dll` into `/usr/lib/x86_64-linux-gnu/wine/` `x86_64-unix/` `x86_64-windows/`
We also drop the files into directories in `/opt/wine-stable/lib64/wine/` and wineasio64.dll into `.wine/drive_c/windows/system32/`.
Now we register:
`wine64 regsvr32 wineasio64.dll`

This worked for me without any problems.
Libraries with kxStudio were not registered.

**Comment by orcun-gokbulut** [comment]
Hi @punkochel, thanks for the info. I have tried your method but failed miserably. ;D
What do you mean by x86_64-unix/ x86_64-windows/ directories ? Are they in /usr/lib ? Can you give us full path?

Thanks

**Comment by punkochel** [comment]
> Hi @punkochel, thanks for the info. I have tried your method but failed miserably. ;D What do you mean by x86_64-unix/ x86_64-windows/ directories ? Are they in /usr/lib ? Can you give us full path?
> 
> Thanks

`wineasio64.so` -> `/usr/lib/x86_64-linux-gnu/wine/x86_64-unix`
`wineasio64.dll` -> `/usr/lib/x86_64-linux-gnu/wine/x86_64-windows`
If you don't have them, then create them.

In the directory `/opt/wine-stable/lib64/wine/` also place the .dll and .so files in the appropriate directories.

**Comment by falkTX** [comment]
issue seems to be that older winegcc added libwine to its linked libraries, which no longer exists in new wine.

this is why the builds from kxstudio repos were broken for systems with new wine, but compiling wineasio manually made it work

luckily I think I found a workaround that works for both, by adding `-nostartfiles` to the winegcc linker flags. this prevents the old winegcc linking against libwine, and still works when building with new winegcc
so hopefully this case gets fixed with bde84a3b8d02112607cfa33f72b51e39b38790a8

I will update the kxstudio packages with this fix-attempt soon.

**Comment by falkTX** [comment]
ok no I was wrong, things work if old libwine.so is still there but fail otherwise.
sadly building stuff with newer wine has become quite the hassle on ubuntu PPAs, due to the removal of i386 packages.

so the kxstudio debs likely will have to be precompiled, maybe I can throw in a docker file to make that reproducible.
